aten/src/ATen/native/native_functions.yaml

```diff
@@ -5100,6 +5100,16 @@
     CUDA: legacy::cuda::_th_std
   supports_named_tensor: True
 
+- func: _amp_unscale_inf_check_(Tensor(a!) self, Tensor inv_scale, Tensor found_inf) -> Tensor(a!)
```

Comment:
Seems like a check for non-finite values occurs BEFORE the unscaling?

Can we call this _amp_non_finite_check_and_unscale_?

Reply:
Greg: found_inf declared this way implies found_inf is const. You need to make found_inf in place and return it. 

Try moving found_inf to just after self and returning it.

Reply:
self and found_inf are both modified in place.  Should I return them both as a tuple, with appropriate aliasing information, ie
```
_amp_unscale_inf_check_(Tensor(a!) self, Tensor(b!) found_inf, Tensor inv_scale) -> (Tensor(a!), Tensor(b!))
```
?